controllers/datadogagent/agent_test.go

```diff
@@ -897,6 +893,26 @@ func defaultSystemProbeEnvVars() []corev1.EnvVar {
 			Name:  datadoghqv1alpha1.DDSystemProbeNPMEnabled,
 			Value: "false",
 		},
+		{
+			Name:  datadoghqv1alpha1.DDSystemProbeConntrackEnabled,
```

Comment:
It means it's no more possible to configure Agent through `datadog.yaml` even if booleans are not set in `DatadogAgent` CRD.
It's not a problem specific to this field, it has been there for a while, but with this PR we should take the time to clarify what we want to support there.

Reply:
Sorry, I don't get this one. Are you talking about the config file of the agent? Do you mean that we don't mirror values to this file?
Happy to chat about how you think we could approach this and use maybe this change to fix it.

Reply:
Any ENV variable that is set by default means that having a configuration in `datadog.yaml` (and nothing set in `DatadogAgent`) is not going to work, the value from `datadog.yaml` being ignored.
As this PR handles defaulting, I think it's a good place to clarify what we want to do for these cases.

Reply:
The goal of this change was to surface defaulted values in the status and stop mutating the spec of the DDA. 
But I agree with you, this problem should be tackled as part of the boarder defaulting initiative.
For the sake of unblocking iterations towards the GA, I am creating a task in our backlog to look into that. 
I still have to work on the `Features`, so all can be done in parallel once we merge this (which is a blocker for the rest).